crates/bevy_asset/src/server/info.rs

```diff
@@ -624,13 +637,22 @@ pub(crate) enum HandleLoadingMode {
 #[error("Cannot allocate a handle because no handle provider exists for asset type {0:?}")]
 pub struct MissingHandleProviderError(TypeId);
 
-fn unwrap_with_context<T>(
-    result: Result<T, MissingHandleProviderError>,
+#[derive(Error, Debug)]
+pub enum GetOrCreateHandleInternalError {
```

Comment:
Doc strings would be nice here? This is kind of a weird error type, and it's not clear how one might extend it or use it elsewhere.